omnigibson/robots/behaviorbot.py

```diff
@@ -0,0 +1,677 @@
+                }
+                for arm_name in self.arm_names
+            }
+        )
+        return controllers
+
+    def load(self):
+        prim = super(Behaviorbot, self).load()
+        for part in self.parts.values():
+            part.load()
+        return prim
+
+    def _post_load(self):
+        super()._post_load()
+
+    def _create_discrete_action_space(self):
+        raise ValueError("Behaviorbot does not support discrete actions!")
+    
+    def update_controller_mode(self):
+        super().update_controller_mode()
+        # overwrite robot params (e.g. damping, stiffess, max_effort) here
+        for link in self.links.values():
+            link.ccd_enabled = True
+        for arm in self.arm_names:
+            for finger_link in self.finger_links[arm]:
+                finger_link.mass = 0.02
+        self.links["base"].mass = 15
+        self.links["lh_palm"].mass = 0.3
+        self.links["rh_palm"].mass = 0.3
+        
+        # set base joint properties
+        for joint_name in self.base_joint_names:
+            self.joints[joint_name].stiffness = 1e8
+            self.joints[joint_name].max_effort = 7500
+
+        # set arm joint properties
+        for arm in self.arm_joint_names:
+            for joint_name in self.arm_joint_names[arm]:
+                self.joints[joint_name].stiffness = 1e6
+                self.joints[joint_name].damping = 1
+                self.joints[joint_name].max_effort = 300
+
+        # set finger joint properties
+        for arm in self.finger_joint_names:
+            for joint_name in self.finger_joint_names[arm]:
+                self.joints[joint_name].stiffness = 1e2
+                self.joints[joint_name].damping = 1
+                self.joints[joint_name].max_effort = 5
+   
+    @property
+    def base_footprint_link_name(self):
+        """
+        Name of the actual root link that we are interested in. 
+        """
+        return "base"
+
+    @property
+    def base_footprint_link(self):
+        """
+        Returns:
+            RigidPrim: base footprint link of this object prim
+        """
+        return self._links[self.base_footprint_link_name]
+
+    def get_position_orientation(self):
+        # If the simulator is playing, return the pose of the base_footprint link frame
+        if self._dc is not None and self._dc.is_simulating():
+            return self.base_footprint_link.get_position_orientation()
+
+        # Else, return the pose of the robot frame
+        else:
+            return super().get_position_orientation()
+
+    def set_position_orientation(self, position=None, orientation=None):
+        super().set_position_orientation(position, orientation)
+        # Move the joint frame for the world_base_joint
+        if self._world_base_fixed_joint_prim is not None:
+            if position is not None:
+                self._world_base_fixed_joint_prim.GetAttribute("physics:localPos0").Set(tuple(position))
```

Comment:
will this work with @cgokmen 's new change that uses the physx backend to query / set poses?

Reply:
confirmed it works